### DISCUSSION TITLE
Validating SCAM-SE

### DISCUSSION THREAD

**Discussion by adamrher** [NCAR/amwg_dev#465] [General]
This thread deals with issues related to the new SCAM-SE https://github.com/ESCOMP/CAM/pull/958

**Comment by adamrher** [comment]
@jtruesdal @PeterHjortLauritzen @JulioTBacmeister @swrneale

**Comment by adamrher** [comment]
I did some runs looking at the gateIII iop in John's PR branch (136_scamse).

![temp_curtain](https://github.com/NCAR/amwg_dev/assets/29961476/45b504cc-1a51-40ed-add0-a3ed27cc15c8)

There's two issues and I can't rule out user error for either at this point. (1) The OMEGA output in the L58dev run is not plotting (more likely this is user error), and (2) that the SCAM-SE runs tend to be 'elongated' in time compared to the EUL runs (I'm more confident this is real). SE and EUL have different ATM_NCPL's, but that shouldn't change the calendar timing of the IOP forcing data, as seems to be occurring here.

ncl script here ```/glade/work/aherring/CESM2/EDMF/scam/curtain_tend.ncl```

**Reply by jtruesdal** [reply]
When I was comparing output I modified the SE coupling time to be the same as EUL.  So they matched up in my case.  I'll take a look and see if I can tell what is going on here.  I haven't tested the 58 level yet.

**Reply by adamrher** [reply]
full disclosure my EUL runs use ATM_NCPL=96 (larger than the 72 default), and SE runs ATM_NCPL=48 (default)

**Reply by jtruesdal** [reply]
I did a quick plot of TDIFF for the se and eul cold start  test01 cases using cam6.  Except for the disparity in the coupling the rest of the namelists looked pretty close.  My guess is that the cases would match better using a similar coupling frequency or even the defaults.  Still the errors are -5 to 5 so not too bad.  I haven't looked at the gate errors before, the arm97 ones look much more similar than these.  
<img width="1023" alt="Screen Shot 2024-01-12 at 1 18 21 AM" src="https://github.com/NCAR/amwg_dev/assets/9287391/7c0f3597-29f2-43f6-b10c-4e393aef98a5">

**Reply by jtruesdal** [reply]
The plot is lame but the run lengths look OK.  There must be an error in your ncl plotting routine. The time samples of your se and eul cases differ in number by a factor of 2 because of the coupling frequencies you've chosen.  In my plots, the very dim yellow spotlights at the top of each plot demarcate 20 days each.  Still these look more different than I would expect.  I'm guessing its partly due to the disparity in coupling.  It could be this case just has more forcing at levels where the model algorithms cause the solution to bifurcate.

On a different tack,  Andrew noted that SCAM's aerosols levels were way off especially high up and he had me put in a relaxation back to monthly means for the aerosol fields.  I do have an update to allow SE nudging but haven't checked that in to your branch yet. The free running errors are a good first look at the model but there is not much to bring SCAM back to reality once its start to diverge, which happens almost immediately in this comparison. Below is my ARM97 TDIFF comparisons (EUL and SE with a 1200 coupling timestep and on the bottom is the SE default 1800 dt). 

![SCAM_ARM97_TDIFF_EUL_vs_SE](https://github.com/NCAR/amwg_dev/assets/9287391/141112fa-9b1e-4dec-8bf3-fea57a41d874)

Without some dynamical feed back the thermal structure can't recover from the accumulated biases and like the 3d model SCAM also has natural tendency to move to its own attractor in state space. The relaxation term on T and/or Q and aerosols helps to make up for the missing dynamic feedback, constraining the solutions in a physically relevant way. Our out of the box configuration uses relaxation.

**Reply by adamrher** [reply]
Hmm ... I'm using the time array in the h0 files which is in units of days, not time-steps, so I don't understand how ncl could not be honoring the variables I'm providing it. Is there a way I can output the time of the iop forcing file that is being interpolated to the current model time-step? Would you be willing to point me where in the code I could add a couple print statements with that info?

**Reply by jtruesdal** [reply]
I checked a few of the h0 files from your script and they appear to written out every timestep.  Maybe I'm looking at the wrong cases though.

**Reply by adamrher** [reply]
Yes they are but the time array is still in days and both ran for 20 days. Most importantly the coordinate variable is still in days:

```
(0)	load:/glade/derecho/scratch/aherring/cam6_3_136_scamse_FSCAM_gateIII_T42_T42_L58_1pes_240110_coldstart_test.001/run/cam6_3_136_scamse_FSCAM_gateIII_T42_T42_L58_1pes_240110_coldstart_test.001.cam.h0.1974-08-30-00000.nc
(1)	load:/glade/derecho/scratch/aherring/cam6_3_136_scamse_FSCAM_gateIII_T42_T42_L58_1pes_240110_coldstart_test.001/run/cam6_3_136_scamse_FSCAM_gateIII_T42_T42_L58_1pes_240110_coldstart_test.001.cam.h0.1974-09-14-00000.nc

Variable: var1
Type: float
Total Size: 245376 bytes
            61344 values
Number of Dimensions: 2
Dimensions and sizes:	[time | 1917] x [lev | 32]
Coordinates: 
            time: [   0..19.95833333333333]
            lev: [3.64346569404006..992.556095123291]
Number Of Attributes: 6
  lon :	 336
  lat :	   9
  mdims :	1
  units :	K/s
  long_name :	T tendency - moist processes
  cell_methods :	time: mean
(0)	load:/glade/derecho/scratch/aherring/cam6_3_136_scamse_FSCAM_gateIII_ne3_ne3_mg37_L58_1pes_240110_coldstart_test.001/run/cam6_3_136_scamse_FSCAM_gateIII_ne3_ne3_mg37_L58_1pes_240110_coldstart_test.001.cam.h0.1974-08-30-00000.nc

Variable: var1
Type: float
Total Size: 122752 bytes
            30688 values
Number of Dimensions: 2
Dimensions and sizes:	[time | 959] x [lev | 32]
Coordinates: 
            time: [   0..19.95833333333333]
            lev: [3.64346569404006..992.556095123291]
Number Of Attributes: 4
  mdims :	1
  units :	K/s
  long_name :	T tendency - moist processes
  cell_methods :	time: mean
```

![temp_curtain](https://github.com/NCAR/amwg_dev/assets/29961476/9d35c9c2-9504-49d2-ae27-d169c294a109)

It could totally still be user error, but I can't seem to figure out why yet.

**Reply by jtruesdal** [reply]
I see now, let me take a look at interpolation.

**Reply by jtruesdal** [reply]
It is a bug, and in the interpolation routine as you surmised.  I've uploaded a fix to the branch. The IOP forcing was not being updated correctly for the case where the forcing is on a shorter timescale than the model time step.  Just a clarification there is no IOP interpolation to the current model time step, it just picks the closest IOP time before the current model time.  If the IOP forcing is days apart SCAM will just hold the forcing constant until the model time is greater than or equal to the next IOP time step.

**Reply by adamrher** [reply]
Thank you for looking into this and resolving it. I realize now this is not related to the new dycore, but rather a long standing bug when the time-step is longer than the iop dt, as u note (the iop dt is 1200 s and I was comparing dtime=900 s to dtime=1800 s). We actually realized this bug in a email thread last year. Ill probably dig it up just to let others on the thread know you've fixed. it.

![temp_curtain](https://github.com/NCAR/amwg_dev/assets/29961476/1c6ce5bf-46d3-4a06-936c-6538761fc4f5)

Thanks again!

**Reply by jtruesdal** [reply]
Thanks for forwarding the thread, new plots and the followup emails.  Hopefully the T and Q tendencies now also show good  agreement between EUL and SE. I'll look into the OMEGA issue next.

**Reply by adamrher** [reply]
Updated the gateIII plot to include the tendencies. There are some differences but I would've expected differences due to the different vertical advection algorithm. And there would also be differences in the solution just from the changing physics time-steps.

To provide some broader context to the OMEGA issue, this cropped up in my pursuit to try to make the FSCAM compset default to CAM7. And I did want to ask you whether this PR would be the the best place to do this. 

For the actual issue of problematic OMEGA output, note that particular run came after a considerable amount of mucking around in the config_comp* files, the FSCAM entries, and I could very well have broken the thing. The run is here, but I will try to recreate it more cleanly when I have time.
```
/glade/derecho/scratch/aherring/cam6_3_136_scamse_FSCAM7_gateIII_ne3_ne3_mg37_L58_1pes_240110_coldstart_test.001
```

**Reply by adamrher** [reply]
@jtruesdal sorry it took all week to re-test this OMEGA thing. Turns out it was me. I can plot OMEGA from the cam.h0 files just fine in the L58 runs. I hope you didn't spend much time working on this.

![temp_curtain](https://github.com/NCAR/amwg_dev/assets/29961476/23318ce7-f8ec-4e22-9009-5a6fa52cfd51)

**Comment by swrneale** [comment]
@jtruesdal @PeterHjortLauritzen @JulioTBacmeister @adamrher  just as an aside here has anyone got the togaII IOP working with a recent tag (#cam6_3_139) on derecho, even with the ice code mod? I can get gateIII to work, but not togaII.

**Reply by jtruesdal** [reply]
From the logs (PET0.ESMF_LogFile and ice.log) it looks like the cice clock is out of sync with the start time for togaII.  I have no idea why these IOPs start in the middle of the day.  I was able to get it to run starting at 0z on the next day.  start_ymd = 19921219 and start_tod=0.  You'll also have to shorten the amount of steps or hours that it runs by 6 hours. Those changes seem to work for me.  I'm not sure what changed in cice that is not allowing it to sync up now.   If it works for you I'll follow up with Dave and maybe Jim Edwards.
